### PR TITLE
Add Changelog.rst

### PR DESCRIPTION
**Pull request by anilbey**
v2.0.0 section of CHANGELOG.rst and the example jupyter notebooks in examples/ directory can help people transition to the new version.

* Added CHANGELOG.rst to list the changes. 
* Filled it with the changes introduced in v2.0.0.
* Provided a link to the CHANGELOG.rst in README.md
* notebooks were slightly out-of-date (af93d63f728d50f638545e629d701b35e564bd1e), added a tox procedure to make sure new changes won't break the example notebooks

### CODE REVIEW

**Comment by anilbey** [comment]
@DrTaDa I basically created the changelog from the confluence link you sent me.

@wvangeit the important changes introduced in BPE2 are listed in the changelog. The examples in /examples dir are using bluepyefe2. The users coming from the legacy version can check them to see how to use bluepyefe2 and read the changelog mentioned in the readme.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+* An automatic step detection has been implemented for some of the simple eCodes (see bluepyefe/ecode/), 
+it provides as an output ton, toff, tend, amp and hypamp. 
+This automatic detection only works when the signal to noise ratio of the stimuli is good enough. 
+Therefore, before exploiting the efeatures, the user should check that the automatic step detection has indeed found the correct step. 
+This can be checked by plotting the real current data on top of the reconstruction of the current resulting from the step detection. 
+For example, if the Extractor instance is called extractor:
+
+
+```python
+for cell in extractor.cells:
```
Comment (DrTaDa) [inline CHANGELOG.rst:35]: That's outdate, the extractor class does not exist anymore.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* changes in plotting function of both features and traces
+
+**Changes in the way the mean and standard deviations are computed**:
+
+* The mean and std for a feature are saved if the number of data point for this target +- tolerance is above the threshold_nvalue_save option. NaNs are not taken into accound when comparing the number of points with threshold_nvalue_save.
+* Any cell for which the rheobase could not be computed is not used when computing the mean or std of the features. In this case, it is however possible to set this rheobase value by hand before calling the mean_efeature function.
+* Instead of saving protocols and efeatures having NaNs as mean or std, BPE2 removes them and issues a warning instead. If for a given protocol/target, there are no non-Nan values, then the protocol is not saved at all (a warning is issued as well).
+
+**Implementation of an automatic step detection when ton/toff/tend/amp is not known**:
+
```
Comment (DrTaDa) [inline CHANGELOG.rst:25]: Maybe add that for non-step protocols, the timings need to be informed by the user

File: CHANGELOG.rst (hunk 1 above)
Comment (anilbey) [reply CHANGELOG.rst:35]: Thanks, I am deleting it.

**Review by wvangeit — APPROVED** [review]

**Pull request merged**